### PR TITLE
{vis}[intel/2016a] p4vasp 0.3.29 (REVIEW)

### PR DESCRIPTION
**Pull request by boegel**
(replacement for #2086)

edit: requires ~~#2628~~, #2631

### CODE REVIEW

**Comment by boegel** [comment]
(continuing where I left off in #2086) Although the build of `p4vasp` works, trying to use it fails on `import gtk` failing.
The output of PyGTK's `./configure` suggests that `gtk+` is missing as a dependency, so I added that, making sure there's no conflict on the `GLib` version.

However, this doesn't work, because `GLib` 2.29.92 is too old for this version of `gtk+` (or at least, for some of its dependencies, e.g. `gdk-pixbuf`).

To be continued...

**Comment by ocaisa** [comment]
I've solved by following what homebrew does, working on the PR now.

**Comment by ocaisa** [comment]
Updated PR for `intel/2016a` at #2443

**Comment by boegel** [comment]
This can be closed as is if #2443 works out... Thanks @ocaisa!

**Comment by boegel** [comment]
Well, not closed, updated to use the `PyGTK` from #2443 :)

**Comment by boegel** [comment]
@wpoely86: please review?

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/37701f96d7440083812f for a full test report.

File: easybuild/easyconfigs/p/p4vasp/p4vasp-0.3.29-intel-2016a-Python-2.7.11.eb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+sanity_check_paths = {
+    'files': ['p4vasp/bin/p4v', 'p4vasp/lib/libODP.a', 'p4vasp/lib/libp4vasp.a'],
+    'dirs': ['p4vasp/data', 'p4vasp/include/ODP', 'p4vasp/include/p4vasp', 'p4vasp/python-packages', 'p4vasp/utils'],
+}
+
+modextrapaths = {
+    'CPATH': ['p4vasp/include'],
+    'LIBRARY_PATH': ['p4vasp/lib'],
+    'PATH': ['p4vasp/bin'],
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/p4vasp/p4vasp-0.3.29-intel-2016a-Python-2.7.11.eb]: aren't these picked up automatically?

Comment (boegel) [inline easybuild/easyconfigs/p/p4vasp/p4vasp-0.3.29-intel-2016a-Python-2.7.11.eb]: no, they're in a `p4vasp` subdir...

**Comment by boegel** [comment]
still several missing pieces: Glade, PyOpenGL, ...

cfr:

```
$ p4v
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/bin/p4v.py", line 225, in createFrame
    xml=p4vasp.util.loadGlade(self.gladefile,self.gladename)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/util.py", line 494, in loadGlade
    import gtk.glade
ImportError: No module named glade
ERROR : Unable to create Glade object from frame.glade, frame_box
Searching for applets
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1, in <module>
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/KpointsViewerApplet.py", line 26, in <module>
    from p4vasp.paint3d.OpenGLPaint3D import *
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/paint3d/OpenGLPaint3D.py", line 5, in <module>
    from OpenGL.GL import *
ImportError: No module named OpenGL.GL
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1, in <module>
ImportError: No module named buggy
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1, in <module>
ImportError: No module named mymod
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1, in <module>
ImportError: No module named foldx
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1
    import 5.3.2.5.3
             ^
SyntaxError: invalid syntax
```

**Comment by ocaisa** [comment]
This will give you the build recipe for glade:
https://github.com/Homebrew/homebrew/blob/749acf67a98627cb0e2bbafe5a07c77a244e9e26/Library/Formula/glade.rb

I think PyOpenGL might be a tough one.

On 12 February 2016 at 08:46, Kenneth Hoste <notifications@github.com<mailto:notifications@github.com>> wrote:

still several missing pieces: Glade, PyOpenGL, ...

cfr:

$ p4v
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/bin/p4v.py", line 225, in createFrame
    xml=p4vasp.util.loadGlade(self.gladefile,self.gladename)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/util.py", line 494, in loadGlade
    import gtk.glade
ImportError: No module named glade
ERROR : Unable to create Glade object from frame.glade, frame_box
Searching for applets
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1, in <module>
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/KpointsViewerApplet.py", line 26, in <module>
    from p4vasp.paint3d.OpenGLPaint3D import *
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/paint3d/OpenGLPaint3D.py", line 5, in <module>
    from OpenGL.GL import *
ImportError: No module named OpenGL.GL
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1, in <module>
ImportError: No module named buggy
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1, in <module>
ImportError: No module named mymod
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1, in <module>
ImportError: No module named foldx
Traceback (most recent call last):
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 106, in findApplets
    o=getClass(c)
  File "/user/scratch/gent/vsc400/vsc40023/easybuild_REGTEST/SL6/sandybridge/software/p4vasp/0.3.29-intel-2016a-Python-2.7.11/p4vasp/python-packages/p4vasp/applet/__init__.py", line 77, in getClass
    exec "import %s\nc=%s\n"%(join(v[:-1],"."),c)
  File "<string>", line 1
    import 5.3.2.5.3
             ^
SyntaxError: invalid syntax


—
Reply to this email directly or view it on GitHub<https://github.com/hpcugent/easybuild-easyconfigs/pull/2328#issuecomment-183219137>.



--
Dr. Alan O'Cais
Application Support
Juelich Supercomputing Centre
Forschungszentrum Juelich GmbH
52425 Juelich, Germany

Phone: +49 2461 61 5213
Fax: +49 2461 61 6656
E-mail: a.ocais@fz-juelich.de<mailto:a.ocais@fz-juelich.de>
WWW:    http://www.fz-juelich.de/ias/jsc/EN


------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------
Forschungszentrum Juelich GmbH
52425 Juelich
Sitz der Gesellschaft: Juelich
Eingetragen im Handelsregister des Amtsgerichts Dueren Nr. HR B 3498
Vorsitzender des Aufsichtsrats: MinDir Dr. Karl Eugen Huthmacher
Geschaeftsfuehrung: Prof. Dr.-Ing. Wolfgang Marquardt (Vorsitzender),
Karsten Beneke (stellv. Vorsitzender), Prof. Dr.-Ing. Harald Bolt,
Prof. Dr. Sebastian M. Schmidt
------------------------------------------------------------------------------------------------
------------------------------------------------------------------------------------------------

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 26 out of 26 (1 easyconfigs in this PR)
Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/173ec612556bd910319b for a full test report.

**Comment by boegel** [comment]
@ocaisa: PyOpenGL was pretty easy (cfr. #2628), Glade is tackled in #2631

**Comment by boegel** [comment]
`p4v` now works on top of PyOpenGL (#2628) and Glade (#2631)... \o/

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2151.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/8d65fe4671a9384c14c9 for a full test report.

**Comment by boegel** [comment]
Jenkins: test this please

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2668.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/941f9f22c06cb581c517 for a full test report.

**Comment by boegel** [comment]
Going in, thanks for the feedback everyone!

**Pull request merged**